engine/flink/scala-utils/src/test/scala/pl/touk/nussknacker/engine/flink/api/typeinfo/caseclass/CaseClassSerializationTest.scala

```diff
@@ -0,0 +1,92 @@
+package pl.touk.nussknacker.engine.flink.api.typeinfo.caseclass
+
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.common.typeinfo.TypeInfo
+import org.apache.flink.api.common.typeutils.TypeSerializer
+import org.apache.flink.api.java.typeutils.TypeExtractor
+import org.apache.flink.core.memory.{DataInputViewStreamWrapper, DataOutputViewStreamWrapper}
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers
+import pl.touk.nussknacker.test.ProcessUtils.convertToAnyShouldWrapper
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import scala.reflect.{ClassTag, classTag}
+
+class CaseClassSerializationTest extends AnyFunSuite with Matchers {
+
+  private val executionConfig = new ExecutionConfig()
+
+  private val bufferSize = 1024
+
+  test("Simple case class should be the same after serialization and deserialization") {
+    val input = SimpleCaseClass("value")
+
+    val serializer   = getSerializer[SimpleCaseClass]
+    val deserialized = serializeAndDeserialize(serializer, input)
+
+    serializer shouldBe a[ScalaCaseClassSerializer[_]]
+    deserialized shouldEqual input
+  }
+
+  test("Case class with field in body should be the same after serialization and deserialization") {
+    val input = SimpleCaseClassWithAdditionalField("value")
+
+    val serializer   = getSerializer[SimpleCaseClassWithAdditionalField]
+    val deserialized = serializeAndDeserialize(serializer, input)
+
+    serializer shouldBe a[ScalaCaseClassSerializer[_]]
+    deserialized shouldEqual input
+  }
+
+  test("Case class with secondary constructor should be the same after serialization and deserialization") {
+    val input = SimpleCaseClassWithMultipleConstructors("value", 2.2)
```

Comment:
WDYT about using the second constructor here?
```
new SimpleCaseClassWithMultipleConstructors(2.2, "value")
```

Reply:
It's a good idea to check it like that. I changed it in the code